hom/utils.py

```diff
@@ -0,0 +1,267 @@
+    existing_ticket_channel = get_user_ticket_channel(interaction.guild, interaction.user)
+
+    if existing_ticket_channel:
+        msg_content = f":envelope:  Click [here]({existing_ticket_channel.jump_url}) to view your open ticket."
+        await interaction.followup.send(content=msg_content, ephemeral=True)
+        return existing_ticket_channel
+
+    channel_name = f"help-{interaction.user.display_name[:15]}"
+    tickets_category = get_category(interaction.guild, "Tickets")  # TODO: Make this channel_id?
+    mod_role = get_role_by_name(interaction.guild, "Moderator")
+    helpful_old_man_role = get_role_by_name(interaction.guild, "Helpful Old Man")
+    assert helpful_old_man_role
+    assert mod_role
+
+    new_text_channel = await interaction.guild.create_text_channel(
+        name=channel_name,
+        category=tickets_category,
+        reason=f"{interaction.user.display_name} ({interaction.user.id}) has opened a ticket.",
+        topic=button_label or "Unknown (This is a bug)",
+        overwrites={
+            interaction.guild.default_role: discord.PermissionOverwrite(
+                read_messages=False,
+                read_message_history=True,
+                attach_files=True,
+                add_reactions=True,
+                embed_links=True,
+            ),
+            mod_role: discord.PermissionOverwrite(read_messages=True),
+            helpful_old_man_role: discord.PermissionOverwrite(read_messages=True),
+            interaction.user: discord.PermissionOverwrite(read_messages=True),
+        },
+    )
+
+    content = (
+        ":envelope:  We have created a support ticket for you, click [here]"
+        f"({new_text_channel.jump_url}) to view."
+    )
+
+    embed = discord.Embed(description=instructions, title=button_label)
+    embed.set_footer(
+        text=(
+            "This channel is only visible to you and our moderators. If your question has been "
+            "answered, feel free to close the ticket."
+        )
+    )
+
+    if example_url:
+        embed.set_image(url=example_url)
+
+    await interaction.followup.send(content, ephemeral=True)
+    await new_text_channel.send(
+        f"{interaction.user.mention}", embed=embed, view=views.SupportMessage()
+    )
+
+    log_content = (
+        f"({new_text_channel.topic}) Ticket opened for user:\n``{interaction.user.display_name}`` "
+        f"- {interaction.user.mention}"
+    )
+
+    assert interaction.client.user
+    await send_log_message(interaction, log_content, interaction.client.user)
+    return new_text_channel
+
+
+async def send_log_message(
+    interaction: discord.Interaction[commands.Bot],
+    content: str,
+    mod: discord.User | discord.ClientUser | discord.Member,
+    channel: t.Optional[discord.TextChannel] = None,
+) -> t.Optional[discord.Message]:
+    assert interaction.guild
+    mod_logs_channel = get_channel_by_name(interaction.client, interaction.guild.id, "mod-logs")
```

Comment:
Same as with the roles, wouldn't it be better to define the channel id in `.env` instead of forcing the name of the channel to `mod-logs`? If the channel name ever changes for any reason, this feature will break.